install_config/persistent_storage/persistent_storage_glusterfs.adoc

```diff
@@ -203,22 +213,198 @@ gluster-default-volume   <none>    2147483648   RWX           Available
-namespace. Any attempt to access a persistent volume across a namespace causes
-the pod to fail.
 
-Additionally, permissions and ownership can be controlled as normal by the
-Gluster server administrators using normal POSIX compliant security.
+This section covers Gluster volume security, including matching permissions and
+SELinux considerations. The reader is expected to understand the basics of POSIX
+permissions, process UIDs, supplemental groups, and SELinux.
+
+[NOTE]
+====
+See
+link:../../install_config/persistent_storage/pod_security_context.html[Volume
+Security] before implementing gluster volumes.
+====
+
+Assume that the target gluster volume, "_HadoopVol_", is mounted under
+_/mnt/glusterfs/_, with the following POSIX permissions and selinux labels:
+
+[[gfs-mount]]
+====
+----
+# ls -lZ /mnt/glusterfs/
+drwxrwx---. yarn hadoop system_u:object_r:fusefs_t:s0    HadoopVol
+
+# id yarn
+uid=592(yarn) gid=590(hadoop) groups=590(hadoop)
+----
+====
+
+In order to access the "_HadoopVol_" volume, containers must match the SELinux
+label, and either run with a UID of 592, or with 590 in their supplemental groups.
+The OpenShift GlusterFS plug-in mounts the volume in the container with the
+same POSIX ownership and permissions found on the target gluster mount, namely
+owner will be 592 and group id will be 590. However, the container is not run with
+its effective UID equal to the owner of the gluster mount, nor with its GID equal
+to the target volume's group id, which is the desired behavior. Instead, container
+UIDs and supplemental groups are controlled by Security Context Constraints
+(SCCs) and by the user's project.
+
+[[gfs-supplemental-groups]]
+=== Group IDs
+
+The recommended way to handle gluster volume access (assuming it is not an option to
+change permissions on the gluster mount) is to use supplemental groups. Supplemental
+groups in OpenShift are used for shared storage, of which GlusterFS is an example. In
+contrast, block storage, such as Ceph RBD or iSCSI, use the *fsGroup* SCC strategy and
+the *fsGroup* value in the pod's `*securityContext*`.
+
+[NOTE]
+====
+It is generally preferable to use supplemental group IDs to gain access to
+persistent storage versus using link:#gfs-user-ids[user IDs]. Supplemental
+groups are covered further in
+link:pod_security_context.html#supplemental-groups[Volume Security].
+====
+
+Because the group ID on the link:#gfs-mount[example target gluster mount] shown
+above is 590, the pod can define that group ID using `*supplementalGroups*`
+under the pod's global `*securityContext*` definition. For example:
+
+====
+----
+spec:
+  containers:
+    - name:
+    ...
+  securityContext: <1>
+    supplementalGroups: [590] <2>
+----
+<1> `*securityContext*` must be defined globally to the pod, not under a
+specific container.
+<2> An array of GIDs defined globally for the pod.
+====
+
+Assuming there are no custom SCCs that might satisfy the pod's requirements, the
```

Comment:
On second thought, this might be the appropriate amount of detail.  I'll think about it, but I think this is okay to leave in for now.  We can massage this throughout the docs in a follow-up if we feel like it is needed.